docs/src/snap/howto/cis-hardening.md

```diff
@@ -282,6 +347,3230 @@ the sets, including the dqlite specific checks in the output.
+For example: `snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--make-iptables-util-chains is not set
+```
+
+##### Control 4.2.8
+
+**Description:**
+
+Ensure that the --hostname-override argument is not set
+(Manual)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file /var/snap/k8s/common/args/kubelet
+on each worker node and remove the --hostname-override
+argument.
+
+Restart the kubelet service.
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--hostname-override is set to false
+```
+
+##### Control 4.2.9
+
+**Description:**
+
+Ensure that the --event-qps argument is set to a level which
+ensures appropriate event capture (Manual)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file /var/snap/k8s/common/args/kubelet on each
+worker node and
+set the --event-qps parameter as appropriate.
+
+Restart the kubelet service.
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--event-qps is not set
+```
+
+##### Control 4.2.10
+
+**Description:**
+
+Ensure that the --tls-cert-file and --tls-private-key-file
+arguments are set as appropriate (Manual)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file /var/snap/k8s/common/args/kubelet on each
+worker node and
+set the following arguments:
+
+```
+--tls-private-key-file=<path/to/private-key-file>
+--tls-cert-file=<path/to/tls-certificate-file>
+```
+
+Restart the kubelet service.
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--tls-cert-file=/etc/kubernetes/pki/kubelet.crt and --tls-
+private-key-file=/etc/kubernetes/pki/kubelet.key
+```
+
+##### Control 4.2.11
+
+**Description:**
+
+Ensure that the --rotate-certificates argument is not set to
+false (Automated)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file /var/snap/k8s/common/args/kubelet on each
+worker node and
+remove the --rotate-certificates=false argument.
+
+Restart the kubelet service.
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+--rotate-certificates is not set
+```
+
+##### Control 4.2.12
+
+**Description:**
+
+Verify that the RotateKubeletServerCertificate argument is
+set to true (Manual)
+
+
+**Remediation:**
+
+Edit the kubelet configuration file /var/snap/k8s/common/args/kubelet on each
+worker node and
+set the argument --feature-
+gates=RotateKubeletServerCertificate=true
+on each worker node.
+
+Restart the kubelet service.
+
+`snap restart k8s.kubelet`
+
+
+**Audit (as root):**
+
+```
+/bin/cat /var/snap/k8s/common/args/kubelet
+```
+
+**Expected output:**
+
+```
+the RotateKubeletServerCertificate feature gate is not set
```

Comment:
expected output does not match remedidation

Reply:
Fixed, see https://github.com/canonical/k8s-snap/pull/802/files#diff-da8b48161c214d6854a37a2ef8a664491413a6b43793ba6e9e6319a49da719ecR3052